### ISSUE TITLE
Transaction failure due to refused connection results in pending task

### ISSUE THREAD

**Issue by OvervCW** [aio-libs/aioredis-py#887]
I'm trying to handle redis connection failure in my code while running a transaction, but I can't figure out how to stop a pending task from being 'leaked'. Consider the following minimal example, using version 1.3.1 of aioredis:

```python
import aioredis
import asyncio

async def test():
    pool = await aioredis.create_redis_pool(
        "redis://localhost",
        timeout=3.0,
        minsize=0,
        maxsize=10,
        loop=asyncio.get_running_loop(),
    )

    try:
        transaction = pool.multi_exec()
        transaction.expireat("foo", 123)
        await transaction.execute()
    except ConnectionRefusedError:
        # handle exception
        pass

asyncio.get_event_loop().run_until_complete(test())
```

If you run this while redis is unavailable (and `ConnectionRefusedError` is triggered), it produces the following message:

```
Task was destroyed but it is pending!
task: <Task pending name='Task-2' coro=<wait_convert() done, defined at /home/user/.local/share/virtualenvs/prototype-Lfy3Uzc6/lib/python3.8/site-packages/aioredis/util.py:58> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f0dec8156d0>()]>>
```

It only seems to happen for commands that contain some kind of conversion at the end. For example, try replacing the `expireat` with `transaction.incrby("foo", 123)` and the message will not appear.

It's possible to fix this in my own code by doing this:

```python
try:
    transaction = pool.multi_exec()
    transaction.expireat("foo", 123)
    await transaction.execute()
except ConnectionRefusedError:
    for fut in transaction._results:
        fut.cancel()

    # handle exception
```

But it seems to me that this is not supposed to be necessary.

**Comment by seandstewart** [comment]
@OvervCW -

We're currently working on a new connections and transactions implementation in #891 .

Could you try out the transactions implementation in that branch?

**Comment by seandstewart** [comment]
This issue should be resolved as of #891 - please feel free to pull in the latest master and test to ensure this is the case.
